### PR TITLE
add apns-collapse-id header to support notification management on iOS10

### PR DESCRIPTION
**Pull request by 123avi**
apns-collapse-id header is used on iOS 10 for notification management.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -57,6 +57,24 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
Comment (jchambers) [inline pom.xml]: Are these changes relevant to the `apns-collapse-id` header? Can we do without them?

File: pushy/src/main/java/com/relayrides/pushy/apns/ApnsClientHandler.java [hunk 2]
```diff
@@ -247,6 +248,16 @@ public void write(final ChannelHandlerContext context, final Object message, fin
                     .path(APNS_PATH_PREFIX + pushNotification.getToken())
                     .addInt(APNS_EXPIRATION_HEADER, pushNotification.getExpiration() == null ? 0 : (int) (pushNotification.getExpiration().getTime() / 1000));
 
+
+
+            if (pushNotification.getCollapseId() != null) {
+                headers.add(APNS_COLLAPSE_ID, pushNotification.getCollapseId());
+            }
+
+            if (pushNotification.getPriority() != null) {
```
Comment (jchambers) [inline pushy/src/main/java/com/relayrides/pushy/apns/ApnsClientHandler.java]: Don't we already have this block on line 261?

File: pushy/src/main/java/com/relayrides/pushy/apns/ApnsPushNotification.java [hunk 3]
```diff
@@ -120,4 +120,23 @@
      * @since 0.5
      */
     String getTopic();
+
+    /**
+     * <p>
+     * Returns the id of the notification. This is generally used for notification management
```
Comment (jchambers) [inline pushy/src/main/java/com/relayrides/pushy/apns/ApnsPushNotification.java]: I think we need to expand/clarify these docs, but I can take responsibility for that.

File: pushy/src/main/java/com/relayrides/pushy/apns/util/SimpleApnsPushNotification.java [hunk 4]
```diff
@@ -77,9 +77,17 @@ public SimpleApnsPushNotification(final String token, final String topic, final
      * @see DeliveryPriority#IMMEDIATE
      */
     public SimpleApnsPushNotification(final String token, final String topic, final String payload, final Date invalidationTime) {
-        this(token, topic, payload, invalidationTime, DeliveryPriority.IMMEDIATE);
+        this(token, topic, payload, invalidationTime, DeliveryPriority.IMMEDIATE,null);
```
Comment (jchambers) [inline pushy/src/main/java/com/relayrides/pushy/apns/util/SimpleApnsPushNotification.java]: Minor, but please add spaces after commas (i.e. change `DeliveryPriority.IMMEDIATE,null` to `DeliveryPriority.IMMEDIATE, null`).

Comment (jchambers) [inline pushy/src/main/java/com/relayrides/pushy/apns/util/SimpleApnsPushNotification.java]: …this applies to line 86, too.

**Comment by jchambers** [comment]
I've left some specific comments inline, but I think we need to address two general things, too:

1. Could you please reopen this pull request against `master`?
2. Tests need to pass before we can merge this. I haven't looked yet, but I suspect the failing tests are a result of the `pom.xml` changes.

**Comment by jchambers** [comment]
Also, where are my manners? Thanks for the contribution!

File: pom.xml (hunk 1 above)
Comment (123avi) [inline pom.xml]: yes, we can , I will remove it . I would suggest to leave the version update of maven-gpg-plugin

File: pushy/src/main/java/com/relayrides/pushy/apns/ApnsClientHandler.java (hunk 2 above)
Comment (123avi) [inline pushy/src/main/java/com/relayrides/pushy/apns/ApnsClientHandler.java]: good catch , thanks , I will fix it

File: pushy/src/main/java/com/relayrides/pushy/apns/ApnsPushNotification.java (hunk 3 above)
Comment (123avi) [inline pushy/src/main/java/com/relayrides/pushy/apns/ApnsPushNotification.java]: thanks, couldn't find actual written something in the docs regarding the apns-collapse-id

File: pushy/src/main/java/com/relayrides/pushy/apns/util/SimpleApnsPushNotification.java (hunk 4 above)
Comment (123avi) [inline pushy/src/main/java/com/relayrides/pushy/apns/util/SimpleApnsPushNotification.java]: 👍

**Comment by 123avi** [comment]
I opened a new pull request against master. so I am closing this one
